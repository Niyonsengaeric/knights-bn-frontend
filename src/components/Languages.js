import { useTranslation } from 'react-i18next';
import React, { useState, useEffect } from 'react';
import '../assets/styles/components/languages.scss';

const Languages = () => {
  const { i18n } = useTranslation();
  const [language, setLanguage] = useState(navigator.language);
  const { t: translate } = useTranslation();

  useEffect(() => {
    i18n.changeLanguage(language);
  }, [language]);

  const handleChange = (e) => {
    const newLang = e.target.value;

    if (newLang !== 'ki' && newLang !== 'en' && newLang !== 'fr') return;
    setLanguage(newLang);
  };

  return (
    <select className="lang lang--list" onChange={(e) => handleChange(e)}>
      <option>
        {translate('Choose language.1')}
      </option>
      <option value="ki">ikinyarwanda </option>
      <option value="en">english</option>
      <option value="fr">fran√ßais</option>
    </select>
  );
};
export default Languages;
