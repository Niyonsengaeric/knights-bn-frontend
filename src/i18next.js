import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import Backend from 'i18next-http-backend';
import LanguageDetector from 'i18next-browser-languagedetector';

i18n
  .use(Backend)
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    fallbackLng: 'en',
    backend: {
      loadPath: '../public/locales/{{lng}}/{{ns}}.json',
    },
    react: {
      useSuspense: false,
    },
    debug: false,
    interpolation: {
      escapeValue: false,
    },
  });


export default i18n;
