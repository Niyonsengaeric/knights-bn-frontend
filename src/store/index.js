import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducers from './reducers/index';

const initialState = {};

const store = createStore(
  reducers,
  initialState,
  composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
