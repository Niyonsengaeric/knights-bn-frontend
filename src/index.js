import React, { Suspense } from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import App from './route/App';
import configureStore from './redux';
import './i18next';

const store = configureStore();

ReactDOM.render(
  <Suspense fallback="Loading ...">
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>
  </Suspense>,
  document.getElementById('root'),
);
