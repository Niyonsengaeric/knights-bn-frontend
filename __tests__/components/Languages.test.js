import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Languages from '../../src/components/Languages';
import '../../src/i18next';

describe('Languages component', () => {
  it('renders correctly', () => {
    const component = renderer
      .create(<Languages />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  it('displays available languages', () => {
    const wrapper = mount(<Languages />);
    expect(wrapper.find('option')).toHaveLength(4);
  });
  it('switches french', () => {
    const handleChange = jest.fn();
    const wrapper = mount(<Languages onChange={(e) => handleChange(e)} />);
    const select = wrapper.find('select');
    expect(select).toExist();

    select.simulate('change', { target: { value: 'fr' } });
  });
  it('switches ikinyarwanda', () => {
    const handleChange = jest.fn();
    const wrapper = mount(<Languages onChange={(e) => handleChange(e)} />);
    const select = wrapper.find('select');
    expect(select).toExist();

    select.simulate('change', { target: { value: 'ki' } });
  });
  it('switches to english ', () => {
    const handleChange = jest.fn();
    const wrapper = mount(<Languages onChange={(e) => handleChange(e)} />);
    const select = wrapper.find('select');
    expect(select).toExist();

    select.simulate('change', { target: { value: 'en' } });
  });
  it('switches language to default', () => {
    const handleChange = jest.fn();
    const wrapper = mount(<Languages onChange={(e) => handleChange(e)} />);
    const select = wrapper.find('select');
    expect(select).toExist();
    select.simulate('change', { target: { value: 'any' } });
  });
});
