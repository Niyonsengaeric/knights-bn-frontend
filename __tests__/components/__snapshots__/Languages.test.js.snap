// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Languages component renders correctly 1`] = `
<select
  className="lang lang--list"
  onChange={[Function]}
>
  <option>
    Choose language.1
  </option>
  <option
    value="ki"
  >
    ikinyarwanda 
  </option>
  <option
    value="en"
  >
    english
  </option>
  <option
    value="fr"
  >
    fran√ßais
  </option>
</select>
`;
